core/src/main/scala/org/broadinstitute/dsde/rawls/entities/local/LocalEntityProvider.scala

```diff
@@ -29,9 +31,34 @@ class LocalEntityProvider(workspace: Workspace, implicit protected val dataSourc
 
   private val workspaceContext = workspace
 
-  override def entityTypeMetadata(): Future[Map[String, EntityTypeMetadata]] = {
+  override def entityTypeMetadata(useCache: Boolean): Future[Map[String, EntityTypeMetadata]] = {
     dataSource.inTransaction { dataAccess =>
-      dataAccess.entityQuery.getEntityTypeMetadata(workspaceContext)
+      dataAccess.workspaceQuery.isEntityCacheDefined(workspaceContext.workspaceIdAsUUID).flatMap { cacheEntriesExistOpt =>
+        val cacheEntriesExist = cacheEntriesExistOpt.getOrElse(false)
```

Comment:
this logic is important. if the user tries to use the cache and it isn't populated, we'll route the request through to the original queries and mark the workspace so it gets picked up by the monitor. this means that some cache requests may still be slow, but only temporarily